pages/provinces/[provinceSlug]/[contactSlug].tsx

```diff
@@ -65,7 +65,7 @@ export const getStaticProps: GetStaticProps = ({ params = {} }) => {
   const province = index ? provinces[index as unknown as number] : null;
   const provinceName = province ? province.name : "";
   const contact =
-    province !== null ? province.data[contactSlug as unknown as number] : null;
+    province !== null ? province.data.find((a) => a.slug == contactSlug) : null;
```

Comment:
Wouldn't the `find` operation here increase the complexity from O(n) to O(n^2) with n = then number of contacts?

We have provinces with hundreds of contacts, but I don't know how much it would slow down our operations here.
Can we use the index instead? Maybe by appending the index to the end of the slug, like what I did with the `provinceSlug`. In fact, that was my reason for appending the index to the `provinceSlug`, so that we can quickly find the province by `provinces[index]` operation with O(n) complexity with n = the number of provinces.

Reply:
Because on the province page first load, it takes some time until the search result appear. I want to make the first data is not from search result, but the original data. But since the contact data in json file is not ordered by `penyedia` name, I changed the fetching process to order it first before saving to file. That's why I didn't use provinces[index] here, because the order is based on `penyedia`.

I changed back the ordering to based on the index so provinces[index] can be used and change the contact list ordering on `getStaticProps`. But seems like the performance is still not optimized.

I may be able to continue this after 4 or 5pm, but if someone can review and try optimize it it'll be really helpful

Reply:
Let's separate this problem into another issue, so I can merge this PR now.
Could you please help us to create an issue for this particular concern, Mbak @ekamuktia? Thanks! 😁 🙏